### PR TITLE
fix: Allow pop up notifications to go under the navigation when scrolling

### PR DESCRIPTION
**Pull request by petesfrench**
## Done

- Lower the z-index of notifications so they go under the navigation
- Put the notifications in their own file

## QA

- go to https://ubuntu-com-14222.demos.haus/#success and scroll down, see that the notification goes under the navigation

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-11728

### CODE REVIEW

**Comment by anthonydillon** [comment]
Thank you, this definitely works but the scroll brings the navigation now over the notification. Could you try adding `scroll-padding-top` to the notifications to try and block this behaviour?

**Comment by petesfrench** [comment]
@pastelcyborg Sure thing. Note that the majority of the changes come from formatting changes applied by Jinja. The scope of the pr has not changed.

I got a little stuck on this and haven't had time to jump back into it. If you want to look into it and see if you can get it working feel free

**Comment by petesfrench** [comment]
@pastelcyborg Excuse my late response. Other things have taken priority over this.
I am not a fan of using fixed position for an element that sticks to our navigation, as the height of the navigation is dynamic on pages such as https://ubuntu.com/desktop. As well, the current implementation of the notification reloads the page of close. I envision a user half-way down the page realizing the popup will persist, closing the popup and then being sent back to the top of the page. Not a great user experience

**Comment by britneywwc** [comment]
Took some time to investigate and here are some updates. To QA:
* Go to landing page, scroll down
* Add #success hash
* See that window scroll does not go to the top of the notification banner
* Repeat with any other page e.g /aws

The issue could be that the notification banner is not registered as part of the page flow. One suggestion could be to move the notification banner from main content wrapper to the navigation header instead. This way, the banner display can persist when user scrolls through the page, without needing to change it to sticky positioning.

File: static/js/src/navigation.js [hunk 1]
```diff
@@ -78,6 +80,13 @@ mainList.addEventListener("click", function (e) {
   }
 });
 
+// The current set up of the meganav doesn't work well with notifications. The simplest fix for this is to close all notifications when a dropdown is clicked. Can be removed on migration to vanilla meganav.
+function closeNotifications() {
+  const notification = document.querySelector(".p-popup-notification:target");
+  console.log(notification);
```
Comment (muhammad-ali-pk) [inline static/js/src/navigation.js:86]:
```suggestion
```

Comment (britneywwc) [inline static/js/src/navigation.js:86]:
```suggestion
```

**Review by muhammad-ali-pk — APPROVED** [review]
@petesfrench Just left a tiny cleanup. The rest is LGTM!

**Review by britneywwc — COMMENTED** [review]
It fixes the issue of weird scrolling behaviour on pages such as /aws#success. Since this is a temporary fix, are we still addressing the scroll to notification & notification display in this PR?

File: static/js/src/navigation.js (hunk 1 above)
Comment (britneywwc) [reply static/js/src/navigation.js:86]: Resolving this as it was already pointed out by Muhammad

**Review by britneywwc — APPROVED** [review]

**Review by britneywwc — APPROVED** [review]

**Pull request merged**